src/utils/work/seekWorkResponse.js

```diff
@@ -1,23 +1,17 @@
 import moment from 'moment';
-import Amplify from '@aws-amplify/core';
-import Storage from '@aws-amplify/storage';
-
 import getAuthJWT from 'utils/getAuthJWT';
 import WorkTimeoutError from 'utils/WorkTimeoutError';
-
+import fetchAPI from 'utils/fetchAPI';
 import unpackResult from 'utils/work/unpackResult';
 import WorkResponseError from 'utils/WorkResponseError';
 
-const seekFromS3 = async (ETag) => {
-  const configuredBucket = Amplify.configure().Storage.AWSS3.bucket;
-  const storageUrl = await Storage.get(
-    ETag,
-    {
-      bucket: configuredBucket.replace('biomage-originals', 'worker-results'),
-    },
-  );
+const seekFromS3 = async (ETag, experimentId) => {
+  const response = await fetchAPI(`/v1/workResults/${experimentId}/${ETag}`);
+
+  const responseJson = await response.json();
```

Comment:
I think you can do `const { signedUrl } = await response.json()`. If `signedUrl` is not defined, its value should be `undefined`, so the following condition works.